src/rust/engine/src/scheduler.rs

```diff
@@ -166,6 +166,14 @@ impl Scheduler {
   }
 }
 
+impl Drop for Scheduler {
+  fn drop(&mut self) {
```

Comment:
From what I can tell, this isn't used yet. Is that intentional?

Reply:
`Drop` is called automatically.

Reply:
Ahhh I see, thank you!!